src/infer_request.cc

```diff
@@ -682,6 +714,72 @@ InferenceRequest::Normalize()
 {
   const inference::ModelConfig& model_config = model_raw_->Config();
 
+  // Fill metadata for raw input
+  if (!raw_input_name_.empty()) {
+    if ((original_inputs_.size() != 1) || (model_config.input_size() != 1)) {
+      return Status(
+          Status::Code::INVALID_ARG,
+          "Raw request must only have 1 input to be deduced but got " +
+              std::to_string(model_config.input_size()) +
+              " inputs for model '" + ModelName() + "'");
+    }
+    auto it = original_inputs_.begin();
+    if (raw_input_name_ != it->first) {
+      return Status(
+          Status::Code::INVALID_ARG,
+          "Unexpected reference name for raw input '" + raw_input_name_ +
+              "' got '" + it->first + "'");
+    }
+    const auto& config_input = model_config.input(0);
+    auto& raw_input = it->second;
+    std::vector<int64_t> shape;
+    if (model_config.max_batch_size() != 0) {
+      shape.emplace_back(1);
+    }
+    int64_t dynamic_axis = -1;
+    size_t element_cnt = 1;
+    for (const auto& dim : config_input.dims()) {
+      if (dim == WILDCARD_DIM) {
+        if (dynamic_axis != -1) {
+          return Status(
+              Status::Code::INVALID_ARG,
+              "The shape of the raw input '" + config_input.name() +
+                  "' can not be deduced because there are more than one "
+                  "variable-sized dimension");
+        }
+        dynamic_axis = shape.size();
+      } else {
+        element_cnt *= (size_t)dim;
+      }
+      shape.emplace_back(dim);
+    }
+    if ((config_input.data_type() == inference::DataType::TYPE_STRING)) {
+      if ((dynamic_axis != -1) || (element_cnt != 1)) {
+        return Status(
+            Status::Code::INVALID_ARG,
+            "For BYTE datatype raw input, the model must have input shape [1]");
+      }
+      // In the case of BYTE data type, we will prepend the byte size to follow
+      // the Triton convention.
+      raw_input_size_ = raw_input.Data()->TotalByteSize();
+      RETURN_IF_ERROR(raw_input.PrependData(
+          &raw_input_size_, sizeof(uint32_t), TRITONSERVER_MEMORY_CPU, 0));
+      // Limit the BYTE raw input not to have host policy specific input for
+      // simplicity, such case won't happen given the current protocol spec.
+      // Will need to extend Input::PrependData() if needed.
+      if (!raw_input.HostPolicyData().empty()) {
+        return Status(
+            Status::Code::INVALID_ARG,
+            "Raw input with host policy specific data is not currently "
```

Comment:
Nit: You may want to remove "specific" to make this clearer. Unless it adds additional meaning.*

*If you want to leave specific in, it's a bit ambiguous. The unambiguous way to write it would be host-policy-specific data, though that's a bit long.